jax/_src/numpy/linalg.py

```diff
@@ -1220,13 +1220,17 @@ def norm(x: ArrayLike, ord: int | str | None = None,
                      " compute a vector-norm, or two axes to compute a matrix-norm.")
 
 @overload
-def qr(a: ArrayLike, mode: Literal["r"]) -> Array: ...
+def qr(a: ArrayLike,
+       mode: Literal["reduced", "complete", "raw", "full"] = "reduced",
+       ) -> QRResult: ...
 @overload
-def qr(a: ArrayLike, mode: str = "reduced") -> Array | QRResult: ...
```

Comment:
I think you will need a fallback overload which takes `mode: str` and returns `QRResult | Array`.

Reply:
Are you sure?  I'm pretty sure that's already subsumed by this annotation.  Maybe you're confusing `mode: str = "reduced"` with `mode: Literal["reduced"]`?